internal/cmd/config/config.go

```diff
@@ -337,12 +339,58 @@ func Parse(d string) (*Config, error) {
 		if !strutil.Printable(result.Worker.Name) {
 			return nil, errors.New("Worker name contains non-printable characters")
 		}
+
 		if result.Worker.TagsRaw != nil {
 			switch t := result.Worker.TagsRaw.(type) {
+			// We allow `tags` to be a simple string containing a reference to
+			// an environment variable (env://) / file path (file://).
+			// We then lookup whatever value is within and parse it.
+			case string:
+				rawTags, err := parseutil.ParsePath(t)
+				if err != nil && !errors.Is(err, parseutil.ErrNotAUrl) {
+					return nil, fmt.Errorf("Error parsing worker tags: %w", err)
+				}
+
+				var temp []map[string]interface{}
+				err = hcl.Decode(&temp, rawTags)
+				if err != nil {
+					return nil, fmt.Errorf("Error decoding raw worker tags: %w", err)
+				}
+
+				if err := mapstructure.WeakDecode(temp, &result.Worker.Tags); err != nil {
+					return nil, fmt.Errorf("Error decoding the worker's tags: %w", err)
+				}
+
 			// HCL allows multiple labeled blocks with the same name, turning it
 			// into a slice of maps, hence the slice here. This format is the
 			// one that ends up matching the JSON that we use in the expression.
 			case []map[string]interface{}:
+				for _, m := range t {
+					for k, v := range m {
+						// We allow the user to pass in only the keys in HCL, and
+						// then set the values to point to an env var/file.
+						valStr, ok := v.(string)
+						if !ok {
+							continue
+						}
+
+						parsed, err := parseutil.ParsePath(valStr)
+						if err != nil {
```

Comment:
Should this be checking:

```go
if err != nil && !errors.Is(err, parseutil.ErrNotAUrl) 
```

Reply:
I removed this due to concerns of getting bad state below but on second thought I think you're right. There's also another check below to prevent bad payloads from getting into the json bit

Reply:
Resolved in 1c2ae70

Reply:
Can you add a test case for this? So something that would have failed with the code prior to 1c2ae70 where `parseutil.ParsePath` returns `parseutil.ErrNotAUrl`?

Reply:
Yep - The problem with that is that the checks behind that error are quite easy to bypass due to the nature of URLs, but probably worth still adding a test case for that

Reply:
Yea, its a strange case, but I think having a test to cover it will help to understand the expected behavior. If it helps here is how `parseutil` unit tests that case: https://github.com/hashicorp/go-secure-stdlib/blob/35ba60c96fcdd1ab809281de44df2240011ec71b/parseutil/parsepath_test.go#L53-L58

Reply:
Yep, that's exactly what I used!

Reply:
Addressed in 9e383a9